cssselect/xpath.py

```diff
@@ -220,7 +220,18 @@ def selector_to_xpath(self, selector, prefix='descendant-or-self::',
         assert isinstance(xpath, self.xpathexpr_cls)  # help debug a missing 'return'
         if translate_pseudo_elements and selector.pseudo_element:
             xpath = self.xpath_pseudo_element(xpath, selector.pseudo_element)
-        return (prefix or '') + _unicode(xpath)
+
+        unicode_xpath = _unicode(xpath)
+        # CSS immediate children (CSS "<> div" to XPath "child::div" or "./div")
+        # Works only at the start of a selector
+        # Needed to get immediate children of a processed selector in Scrapy
+        # product = response.css('.product')
+        # name = product.css('<> div')
+        child_re = r'^[ \t\r\n\f]*\<[ \t\r\n\f]*\/'
```

Comment:
precompile the pattern outside of the hot path. A class level attribute is fine.

Reply:
Understood, thanks :)

Reply:
Fixed.

Reply:
Fixed :)